src/main/java/io/ballerina/plugins/idea/preloading/TerminatorUnix.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.ballerina.plugins.idea.preloading;
+
+import org.apache.commons.compress.utils.IOUtils;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+
+import static io.ballerina.plugins.idea.BallerinaConstants.BAL_LOG_PREFIX;
+
+/**
+ * Launcher Terminator Implementation for Unix.
+ *
+ * @since 2.0.0
+ */
+public class TerminatorUnix extends Terminator {
+
+    /**
+     * Get find process command.
+     *
+     * @param script absolute path of ballerina file running
+     * @return find process command
+     */
+    private String[] getFindProcessCommand(String script) {
+
+        String[] cmd = {
+                "/bin/sh", "-c", "ps ax | grep " + script + " |  grep -v 'grep' | awk '{print $1}'"
+        };
+        return cmd;
+    }
+
+    public void terminate() {
```

Comment:
This code segment is copied among all the `Terminator` implementations. Can't we do like this ?
```java
abstract class Terminator {

// other codes
    public void terminate() {
        executeTerminate(LS_CMD_PROCESS_ID);
    }

    // this should be implemented in all the `Terminator` implementations.
    abstract void executeTerminate(string processName);
}
```

Reply:
@ayeshLK as per https://github.com/ballerina-platform/plugin-intellij/pull/49#discussion_r1590197836, we are planning to remove the entire terminator implementation from the new code base.

Reply:
@NipunaRanasinghe yes I understand that. But, here the point is we are duplicating the same code in different places which breaks DRY principle. I think it is better to find the proper abstraction for the `Terminator` functionality and combine the shared logic into one place and provide extension points for the varying parts.

Reply:
Ack